substratevm/src/com.oracle.svm.agent.core/src/com/oracle/svm/agent/core/JvmtiAgentBase.java

```diff
@@ -0,0 +1,186 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.agent.core;
+
+import static com.oracle.svm.agent.core.Support.check;
+import static com.oracle.svm.agent.core.Support.checkJni;
+import static com.oracle.svm.agent.core.Support.fromCString;
+import static com.oracle.svm.agent.core.jvmti.JvmtiEvent.JVMTI_EVENT_THREAD_END;
+import static com.oracle.svm.agent.core.jvmti.JvmtiEvent.JVMTI_EVENT_VM_DEATH;
+import static com.oracle.svm.agent.core.jvmti.JvmtiEvent.JVMTI_EVENT_VM_INIT;
+import static com.oracle.svm.agent.core.jvmti.JvmtiEvent.JVMTI_EVENT_VM_START;
+import static com.oracle.svm.agent.core.jvmti.JvmtiEventMode.JVMTI_ENABLE;
+import static com.oracle.svm.jni.JNIObjectHandles.nullHandle;
+import static org.graalvm.word.WordFactory.nullPointer;
+
+import org.graalvm.nativeimage.CurrentIsolate;
+import org.graalvm.nativeimage.StackValue;
+import org.graalvm.nativeimage.UnmanagedMemory;
+import org.graalvm.nativeimage.c.function.CEntryPoint;
+import org.graalvm.nativeimage.c.function.CEntryPointLiteral;
+import org.graalvm.nativeimage.c.function.CFunctionPointer;
+import org.graalvm.nativeimage.c.struct.SizeOf;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.WordPointer;
+import org.graalvm.word.PointerBase;
+
+import com.oracle.svm.agent.core.jvmti.JvmtiEnv;
+import com.oracle.svm.agent.core.jvmti.JvmtiEventCallbacks;
+import com.oracle.svm.core.c.function.CEntryPointOptions;
+import com.oracle.svm.core.c.function.CEntryPointSetup;
+import com.oracle.svm.core.util.VMError;
+import com.oracle.svm.jni.nativeapi.JNIEnvironment;
+import com.oracle.svm.jni.nativeapi.JNIErrors;
+import com.oracle.svm.jni.nativeapi.JNIJavaVM;
+import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
+import com.oracle.svm.jni.nativeapi.JNIVersion;
+
+public abstract class JvmtiAgentBase<T extends HandleSet> {
```

Comment:
Where's the javadoc comment explaining the purpose and usage of this class?

Reply:
I will add documentation for this class.